os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -151,9 +151,15 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 #ifdef CONFIG_APP_BINARY_SEPARATION
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	ARCH_GET_RET_ADDRESS
-	loadinfo->textalloc = (uintptr_t)mm_malloc(loadinfo->uheap, textsize + datasize, retaddr);
+	loadinfo->textalloc = (uintptr_t)mm_memalign(loadinfo->uheap, textsize, textsize, retaddr);
+	ARCH_GET_RET_ADDRESS
+	loadinfo->roalloc = (uintptr_t)mm_memalign(loadinfo->uheap, loadinfo->rosize, loadinfo->rosize, retaddr);
+	ARCH_GET_RET_ADDRESS
+	loadinfo->dataalloc = (uintptr_t)mm_malloc(loadinfo->uheap, datasize, retaddr);
```

Comment:
Is it okay not to make alignment for data region?

Reply:
Yes. We are making separate MPU regions for text and ro.  The data region will come under the default MPU region which covers the whole partition. So, we dont have to align data region.